### ISSUE TITLE
Issues with TypeScript Compatibility in @maptiler/sdk 2.3.0 and Related Packages

### ISSUE THREAD

**Issue by StazriN** [maptiler/maptiler-sdk-js#122]
I recently updated to `@maptiler/sdk 2.3.0` and encountered a few TypeScript-related issues while working with `@maptiler/weather 2.1.0` and `@maptiler/marker-layout 1.0.1`. Below are the details of the issues I faced:

### 1. Incompatibility with Weather Layers (`@maptiler/weather 2.1.0`)

When trying to add weather layers such as `TemperatureLayer`, `RadarLayer`, `WindLayer`, and `PrecipitationLayer` via `Map.addLayer()`, I get the following TypeScript error:

```typescript
Argument of type 'TemperatureLayer | RadarLayer | WindLayer | PrecipitationLayer' is not assignable to parameter of type 'CustomLayerInterface | (LayerSpecification & { source?: string | SourceSpecification | undefined; })'.

Type 'TemperatureLayer' is not assignable to type 'CustomLayerInterface | (LayerSpecification & { source?: string | SourceSpecification | undefined; })'.
```

It appears that the provided weather layers are not compatible with the expected `CustomLayerInterface` or `LayerSpecification` types.

**Steps to Reproduce:**

1. Install `@maptiler/sdk 2.3.0` and `@maptiler/weather 2.1.0`.
2. Attempt to add a weather layer using `Map.addLayer()`.

**Expected Behavior:**
The weather layers should be added without TypeScript compatibility issues.

**Actual Behavior:**
TypeScript throws an incompatibility error, indicating that the `TemperatureLayer` and other weather layers do not conform to the expected parameter type.

### 2. MarkerLayout Map Type Incompatibility (`@maptiler/marker-layout 1.0.1`)

The second issue I encountered is with the `MarkerLayout` class from `@maptiler/marker-layout 1.0.1`. When initializing `MarkerLayout` with a `Map` instance, I receive the following TypeScript error:

```typescript
Argument of type 'import(".../node_modules/@maptiler/sdk/dist/src/Map").Map' is not assignable to parameter of type 'import(".../node_modules/@maptiler/marker-layout/node_modules/@maptiler/sdk/dist/Map").Map'.

Types have separate declarations of a private property 'isTerrainEnabled'.
```

It seems like there are multiple `Map` types being used due to different package versions, causing the incompatibility.

**Steps to Reproduce:**

1. Install `@maptiler/sdk 2.3.0` and `@maptiler/marker-layout 1.0.1`.
2. Create a new instance of `MarkerLayout` with a `Map` instance.

**Expected Behavior:**
`MarkerLayout` should accept the `Map` instance without any type issues.

**Actual Behavior:**
TypeScript throws an error indicating that the `Map` types are incompatible due to different versions or declarations of private properties.

### Environment

- **@maptiler/sdk**: 2.3.0
- **@maptiler/weather**: 2.1.0 (uses @maptiler/sdk 2.2.2 in package.json)
- **@maptiler/marker-layout**: 1.0.1 (uses @maptiler/sdk 2.1.0 in package.json)
- **TypeScript**: 5.x

### Additional Context

It looks like these issues are due to type mismatches or discrepancies between different versions of the packages. Specifically, `@maptiler/marker-layout` uses `@maptiler/sdk 2.1.0` and `@maptiler/weather` uses `@maptiler/sdk 2.2.2` in their respective `package.json` files, while I'm using `@maptiler/sdk 2.3.0`. It might help if the `@maptiler` team updates the `@maptiler/sdk` version used in their other libraries to match the latest version to ensure compatibility. Any help or guidance on resolving these issues would be greatly appreciated.

**Comment by jonathanlurie** [comment]
Hello @StazriN Thanks a lot for your feedback. Indeed, since the last version, Maplibre has changed the custom layer interface signature. It consist mainly in adding optional args so it's should not cause any issue at runtime, but I'll try to be quick at updating the Marker-layout and Weather lib accordingly! I'll let you know when the new versions are on NPM.

**Comment by jonathanlurie** [comment]
Hello again @StazriN , I just tried it and it turns out your issue may be very closely related to this recent one https://github.com/maptiler/maptiler-sdk-js/issues/117

It comes from Maplibre types not being fully visible from your project. Can you please try to add this to your `tsconfig`:
```ts
{
  "compilerOptions": {
    // ...
    "esModuleInterop": true,
  }
}
```

The issue pops up because Maplibre GL JS is exposed as a CommonJS bundle and this option set as `true` increases compatibility between different bundling strategies ( [read more about it here](https://www.typescriptlang.org/tsconfig/#esModuleInterop) ).

Please let me know if that helps. BTW, this info is going to be part of the MapTiler doc very soon.

**Comment by jonathanlurie** [comment]
As per `@maptiler/marker-layout`, I could not reproduce the issue either, even with the package depending on an older version of the SDK. Can you please give more info about your bundler and configuration? As somewhat of a reference, I tested this on a starter I've made https://github.com/jonathanlurie/maptiler-sdk-vite-vanilla-ts

**Comment by jonathanlurie** [comment]
That's great news! I am less familiar with Yarn (sticking to NPM) so I must admit this option did not come to my mind!
In the process, I just released `@maptiler/marker-layout@1.2.0` , which relies on SDK 2.3.0
